client/src/components/Layout.js

```diff
@@ -29,8 +30,19 @@ const ProgressBar = styled(PageLoader)`
 `
 
 export const Layout = ({ children }) => {
-  const { bannerHeight } = useBanner()
   const router = useRouter()
+  const { stickyBannerHeight } = useStickyBanner()
```

Comment:
I suggest that we instead:
- track updates to fixed header on open/close of sticky banner
- move contribute banner inside of main
- add support for named banners
- have only one banner that takes children to render contents, a name and a flag for fullWidth
- move the nf / contribute content into named banner components ie `ContributeBanner.js` and pass that as a child into `Banner`

Reply:
Hello @davidsmejia, I updated the following items and now the code is ready for your review 👀 
>     * track updates to fixed header on open/close of sticky banner
Instead, I incorporated the [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) Web API(as a hook - see the resource included in the comment). It let us track DOM element object changes with no additional checks and also circumvent the `useRef` limitation to detect DOM changes

>     * move contribute banner inside of main
I discovered that our current structure's `main` is set to fixed width (not full), and the banner needs to be extended as full width, so it remained in place (otherwise we could u[date the site structure, but that could require another PR!)

>     * add support for named banners
>     * have only one banner that takes children to render contents, a name and a flag for fullWidth
>     * move the nf / contribute content into named banner components ie `ContributeBanner.js` and pass that as a child into `Banner`
Now it takes the unique`id` and `fullWidth` props and takes `children`, and created named banner components.